### PR TITLE
Added windows pytest

### PR DESCRIPTION
**Pull request by PalashGoiporia**

### CODE REVIEW

**Comment by vtomole** [comment]
Please solve the conflict.

**Comment by PalashGoiporia** [comment]
@vtomole not sure how to deal with the permission error. It passes locally.

**Comment by vtomole** [comment]
@PalashGoiporia  You need to run the formatter.

**Comment by vtomole** [comment]
Won't merge until @richrines1 approves.

**Comment by PalashGoiporia** [comment]
Ready for review @vtomole @richrines1

File: applications_superstaq/user_config_test.py [hunk 1]
```diff
@@ -44,15 +44,16 @@ def test_service_aqt_upload_configs(mock_aqt_compile: mock.MagicMock) -> None:
     )
     service = applications_superstaq.user_config.UserConfig(client)
 
-    with open("/tmp/Pulses.yaml", "w") as pulses_file:
+    pulse = tempfile.NamedTemporaryFile("w+", delete=False)
```
Comment (vtomole) [inline applications_superstaq/user_config_test.py:47]: Revert from `w+` to `w`

File: applications_superstaq/user_config_test.py [hunk 2]
```diff
@@ -44,15 +44,16 @@ def test_service_aqt_upload_configs(mock_aqt_compile: mock.MagicMock) -> None:
...
+    variable = tempfile.NamedTemporaryFile("w+", delete=False)
```
Comment (vtomole) [inline applications_superstaq/user_config_test.py:48]: Revert from `w+` to `w`

File: applications_superstaq/user_config_test.py [hunk 3]
```diff
@@ -65,31 +66,27 @@ def test_service_aqt_get_configs(mock_aqt_compile: mock.MagicMock) -> None:
     )
     service = applications_superstaq.user_config.UserConfig(client)
 
-    pulses_file = secrets.token_hex(nbytes=16)
-    variables_file = secrets.token_hex(nbytes=16)
+    pulses_file = tempfile.NamedTemporaryFile("w+", delete=False)
+    variables_file = tempfile.NamedTemporaryFile("w+", delete=False)
```
Comment (vtomole) [inline applications_superstaq/user_config_test.py:70]: Revert from `w+` to `w`

**Review by vtomole — CHANGES_REQUESTED** [review]

File: applications_superstaq/user_config_test.py [hunk 4]
```diff
@@ -65,31 +66,27 @@ def test_service_aqt_get_configs(mock_aqt_compile: mock.MagicMock) -> None:
     )
     service = applications_superstaq.user_config.UserConfig(client)
 
-    pulses_file = secrets.token_hex(nbytes=16)
-    variables_file = secrets.token_hex(nbytes=16)
+    pulses_file = tempfile.NamedTemporaryFile("w+", delete=False)
```
[the remark this answers is not in the document]
Comment (PalashGoiporia) [reply applications_superstaq/user_config_test.py:69]: I think this one is needed

File: applications_superstaq/user_config_test.py (hunk 3 above)
Comment (PalashGoiporia) [reply applications_superstaq/user_config_test.py:70]: I think this one is needed

File: applications_superstaq/user_config_test.py [hunk 5]
```diff
@@ -43,16 +44,19 @@ def test_service_aqt_upload_configs(mock_aqt_compile: mock.MagicMock) -> None:
...
     service = applications_superstaq.user_config.UserConfig(client)
-
-    with open("/tmp/Pulses.yaml", "w") as pulses_file:
+    tempdir = tempfile.gettempdir()
+    pulse = secrets.token_hex(nbytes=16)
+    variable = secrets.token_hex(nbytes=16)
+    with open(f"{tempdir}/{pulse}.yaml", "w") as pulses_file:
         pulses_file.write("Hello")
+    with open(f"{tempdir}/{variable}.yaml", "w") as variables_file:
+        variables_file.write("Hello")
```
Comment (vtomole) [inline applications_superstaq/user_config_test.py:53]: Revert variables to write `World`.

**Review by vtomole — APPROVED** [review]

**Comment by singular-value** [comment]
billgates

**Pull request merged**